src/test/java/book/project/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,284 @@
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+        tearDown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/categories/delete-for-category-tests.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/add-three-books.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/add-two-categories.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        tearDown(dataSource);
+    }
+
+    @SneakyThrows
+    static void tearDown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(false);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/delete-all-books.sql")
+            );
+        }
+    }
+
+    @Test
+    @Sql(
+            scripts = "classpath:database/books/delete-title-book.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Create a new book")
+    void createBook_ValidRequestDto_Success() throws Exception {
+        //Given (Arrange)
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setTitle("Title")
+                .setAuthor("Author")
+                .setIsbn("1234567890123")
+                .setPrice(BigDecimal.valueOf(14.99))
+                .setCategoryIds(Set.of(1L));
+
+        BookDto bookDto = new BookDto()
+                .setTitle(requestDto.getTitle())
+                .setAuthor(requestDto.getAuthor())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice())
+                .setCategoryIds(requestDto.getCategoryIds());
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        //When (Act)
+        MvcResult result = mockMvc.perform(post("/books")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        //Then (Assert)
+        BookDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), BookDto.class);
+
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(bookDto, actual, "id");

```

Comment:
This method returns a boolean and doesn’t work as you expected without assertTrue (fix it everywhere)